### PR TITLE
Elasticsearch 5 support

### LINKED ISSUE #3130: Elasticsearch 5 support
When I execute `./manage.py update_index --settings=web6c.settings.production`, I was get some Exception.

> The [string] type is removed in 5.0 and automatic upgrade failed because parameters [index_analyzer, boost] are not supported for automatic upgrades. You should now use either a [text] or [keyword] field instead for field [name]

But I never used elasticsearch.

> Traceback (most recent call last):
  File "./manage.py", line 12, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/lib/python3.5/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/usr/lib/python3.5/site-packages/django/core/management/__init__.py", line 359, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/lib/python3.5/site-packages/django/core/management/base.py", line 294, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/lib/python3.5/site-packages/django/core/management/base.py", line 345, in execute
    output = self.handle(*args, **options)
  File "/usr/lib/python3.5/site-packages/wagtail/wagtailsearch/management/commands/update_index.py", line 120, in ha
ndle
    self.update_backend(backend_name, schema_only=options['schema_only'])
  File "/usr/lib/python3.5/site-packages/wagtail/wagtailsearch/management/commands/update_index.py", line 77, in upd
ate_backend
    index.add_model(model)
  File "/usr/lib/python3.5/site-packages/wagtail/wagtailsearch/backends/elasticsearch.py", line 513, in add_model
    index=self.name, doc_type=mapping.get_document_type(), body=mapping.get_mapping()
  File "/usr/lib/python3.5/site-packages/elasticsearch/client/utils.py", line 71, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/usr/lib/python3.5/site-packages/elasticsearch/client/indices.py", line 282, in put_mapping
    '_mapping', doc_type), params=params, body=body)
  File "/usr/lib/python3.5/site-packages/elasticsearch/transport.py", line 327, in perform_request
    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
  File "/usr/lib/python3.5/site-packages/elasticsearch/connection/http_urllib3.py", line 124, in perform_request
    self._raise_error(response.status, raw_data)
  File "/usr/lib/python3.5/site-packages/elasticsearch/connection/base.py", line 122, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
elasticsearch.exceptions.RequestError: TransportError(400, 'illegal_argument_exception', 'The [string] type is removed in 5.0 and automatic upgrade failed because parameters [index_analyzer, boost] are not supported for automatic upgrades. You should now use either a [text] or [keyword] field instead for field [name]')

### PR DESCRIPTION
**Pull request by kaedroho**
Fixes #3130

This pull request adds a new ``elasticsearch5`` backend into Wagtail which is compatible with Elasticsearch 5.x.

### CODE REVIEW

File: wagtail/wagtailsearch/tests/test_elasticsearch2_backend.py [hunk 1]
```diff
@@ -433,7 +433,7 @@ def test_isnull_false_lookup(self):
         # Check it
         expected_result = {'filtered': {'filter': {'and': [
             {'match': {'content_type': 'searchtests.SearchTest'}},
-            {'not': {'missing': {'field': 'title_filter'}}}
+            {'exists': {'field': 'title_filter'}}
```
Comment (tomdyson) [inline wagtail/wagtailsearch/tests/test_elasticsearch2_backend.py]: Why have the ES2 tests changed?

Comment (kaedroho) [inline wagtail/wagtailsearch/tests/test_elasticsearch2_backend.py]: See: https://github.com/torchbox/wagtail/pull/3145/commits/b8f267aa65e0dfb3a86e1aefbbaa370610dd568b (this one affects all backends)

**Comment by tomdyson** [comment]
@m1kola can you review please?

**Comment by m1kola** [comment]
Yes, I can. I'll take a look at it on weekend.

**Comment by m1kola** [comment]
Something wrong with tests. Elasticsearch5 can't start. It's hard to say what exactly goes wrong, because we only have the following in the build log:
```
Setting up elasticsearch (5.0.0-alpha5) ...
 * Starting Elasticsearch Server       fail]
```

Except tests, looks good to me.

Thanks, Karl!

File: scripts/travis/install_elasticsearch5.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+sudo apt-get autoremove --purge elasticsearch
+wget -qO - https://packages.elastic.co/GPG-KEY-elasticsearch | sudo apt-key add -
+echo "deb http://packages.elastic.co/elasticsearch/5.x/debian stable main" | sudo tee -a /etc/apt/sources.list.d/elk.list
```
Comment (tomdyson) [inline scripts/travis/install_elasticsearch5.sh]: These look like the wrong URLs. From https://www.elastic.co/guide/en/elasticsearch/reference/current/deb.html#deb-repo:

```
wget -qO - https://artifacts.elastic.co/GPG-KEY-elasticsearch | sudo apt-key add -
echo "deb https://artifacts.elastic.co/packages/5.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-5.x.list
```

**Comment by qwIvan** [comment]
thanks

**Comment by kaedroho** [comment]
Elasticsearch 5 tests now working on Travis!

**Comment by gasman** [comment]
Merged in 5df8f1e34efc9bf1ce9790a1fa359bb1a549c143 + parents.

**Pull request closed without merging**